### PR TITLE
sso: utilize secondary emails for existing accounts

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
err fuuu git pull-request using wrong title

File: src/sentry/auth/helper.py [hunk 1]
```diff
@@ -391,6 +391,15 @@ def _get_display_name(self, identity):
     def _get_identifier(self, identity):
         return identity.get('email') or identity.get('id')
 
+    def _find_existing_user(self, email):
+        return User.objects.filter(
+            id__in=UserEmail.objects.filter(
+                email__iexact=email,
+                is_verified=True,
```
Comment (dcramer) [inline src/sentry/auth/helper.py]: this is a slight change in behavior in that:

1. it doesnt prioritize username
2. it only allows merge with verified emails (this is actually something we should have done immediately from a security POV)

Comment (bretthoerner) [inline src/sentry/auth/helper.py]: > it doesnt prioritize username

It's not checking by username at all now, is it?

**Review by bretthoerner — CHANGES_REQUESTED** [review]

**Comment by dcramer** [comment]
@bretthoerner its not, but the username SHOULDNT matter, as if its an email, it should also be a registered UserEmail, and we'd still want the is_verified check

**Review by bretthoerner — APPROVED** [review]

**Pull request merged**